pkg/storage/tsdb/block/meta.go

```diff
@@ -44,7 +45,8 @@ const (
 type Meta struct {
 	tsdb.BlockMeta
 
-	Thanos ThanosMeta `json:"thanos"`
+	Thanos       ThanosMeta `json:"thanos"`
+	LastModified *time.Time `json:"last_modified,omitempty"`
```

Comment:
We're adding field to meta.json that will only be set and used by compactor -- in one specific case. Furthermore it will not reflect any possible modification of meta.json file in the bucket. This looks like a recipe for confusion to me.

Reply:
The intent is that this field would only be populated in block meta on compactors and in the meta.json files they cache locally. It would not be populated in object storage. The reason we even store this field in the meta.json is because that file is cached on compactors. 

One alternative would be to always read the LastModified time from object storage, even if we use a cached meta.json, but that's not ideal. Another alternative would be to use the ULID time instead of LastModified time for this metric, but it's less accurate. Thoughts on those?

Reply:
> The reason for even storing this field in the meta.json is because that file is cached on compactors and it would be nice to avoid reading the object attributes when we're using the cache.

We're saving literally one call to fetch attributes per block (because block can only be compacted once). I don't think it's worth the effort.

Reply:
Ok, switched to fetching attributes for all compactor usage.